bin/report

```diff
@@ -7,22 +7,154 @@ set -o errexit    # always exit on error
+kv_pair_string "npm_version_request" "$(read_json "$BUILD_DIR/package.json" ".engines.npm")"
+kv_pair_string "yarn_version_request" "$(read_json "$BUILD_DIR/package.json" ".engines.yarn")"
+
+# build scripts
+kv_pair_string "start_script" "$(read_json "$BUILD_DIR/package.json" ".scripts[\"start\"]")"
+kv_pair_string "build_script" "$(read_json "$BUILD_DIR/package.json" ".scripts[\"build\"]")"
+kv_pair_string "postinstall_script" "$(read_json "$BUILD_DIR/package.json" ".scripts[\"postinstall\"]")"
+kv_pair_string "heroku_prebuild_script" "$(read_json "$BUILD_DIR/package.json" ".scripts[\"heroku-prebuild\"]")"
+kv_pair_string "heroku_postbuild_script" "$(read_json "$BUILD_DIR/package.json" ".scripts[\"heroku-postbuild\"]")"
+kv_pair "build_script_time" "$(meta_get "build-script-time")"
+kv_pair "build_script_memory" "$(meta_get "build-script-memory")"
+kv_pair "postinstall_script_time" "$(meta_get "postinstall-script-time")"
+kv_pair "postinstall_script_memory" "$(meta_get "postinstall-script-memory")"
+kv_pair "heroku_prebuild_script_time" "$(meta_get "heroku-prebuild-script-time")"
+kv_pair "heroku_prebuild_script_memory" "$(meta_get "heroku-prebuild-script-memory")"
+kv_pair "heroku_postbuild_script_time" "$(meta_get "heroku-postbuild-script-time")"
+kv_pair "heroku_postbuild_script_memory" "$(meta_get "heroku-postbuild-script-memory")"
+
+# which package manager was used
+if [[ "$YARN" == "true" ]]; then
+  kv_pair_string "package_manager" "yarn"
 else
-  node_version=""
+  kv_pair_string "package_manager" "npm"
 fi
 
-[ -f "$BUILD_DIR/yarn.lock" ] && package_manager="yarn" || package_manager="npm"
+# We have to get this info from bin/compile since a lockfile will be generated if there wasn't one
+kv_pair "lock_file" "$(meta_get "has-node-lock-file")"
+
+# does this project use "workspaces"?
+kv_pair "uses_workspaces" "$(json_has_key "$BUILD_DIR/package.json" "workspaces")"
+# what workspaces are defined? Logs as: `["packages/*","a","b"]`
+kv_pair_string "workspaces" "$(read_json "$BUILD_DIR/package.json" ".workspaces")"
+# count # of js, jsx, ts, coffee, vue, and html files to approximate project size, exclude any files in node_modules
+kv_pair "num_project_files" "$(find "$BUILD_DIR" -name '*.js' -o -name '*.ts' -o -name '*.jsx' -o -name '*.coffee' -o -name '*.vue' -o -name '*.html' | grep -cv node_modules | tr -d '[:space:]')"
+# measure how large node_modules is on disk
+kv_pair "node_modules_size" "$(measure_size)"
+
+# pull metadata from the build
+kv_pair_string "build_step" "$(meta_get "build-step")"
+kv_pair_string "failure" "$(meta_get "failure")"
+kv_pair_string "cache_status" "$(meta_get "cache-status")"
+kv_pair "build_time" "$(meta_get "build-time")"
+kv_pair "skipped_prune" "$(meta_get "skipped-prune")"
+kv_pair "cached_bower_components" "$(meta_get "cached-bower-components")"
+kv_pair "custom_cache_dirs" "$(meta_get "node-custom-cache-dirs")"
+
+# pull execution data from the build
```

Comment:
All of this timing data would probably be better captured in a trace, but we can iterate towards that